snuba/request_schema.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import copy
+import itertools
+import jsonschema
+
+from collections import ChainMap
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+from snuba.query.extensions import QueryExtension
+from snuba.query.query import Query
+from snuba.schemas import Schema, GENERIC_QUERY_SCHEMA
+
+
+@dataclass(frozen=True)
+class Request:
+    query: Query
+    extensions: Mapping[str, Mapping[str, Any]]
```

Comment:
I'm curious what your thoughts are on the evolution on `extensions` here — it seems like `Mapping[str, Any]` won't make sense as the value type of this mapping for much longer? I guess we don't have to worry about this until the `body` property below is removed, though.

Reply:
I did not put too much thinking into this.
I guess it should be something with better typing but such that can be parsed from json fairly directly, otherwise we would have another query processing phase before query processing.
I think a typed dict or a dataclass could work.

Reply:
@tkaemming, about the idea we discussed today (removing this extensions content and make the extension take a query and produce a new query parsing the extension according to the schema so that we can have proper types). I would do that in a separate PR, this one would become quite a bit more complex (I would have to unpack the RequestSchema.validate method and split its content).